pallets/msa/src/lib.rs

```diff
@@ -511,26 +548,8 @@ pub mod pallet {
 		#[pallet::call_index(2)]
 		#[pallet::weight(T::WeightInfo::create_provider(provider_name.len() as u32))]
 		pub fn create_provider(origin: OriginFor<T>, provider_name: Vec<u8>) -> DispatchResult {
-			let provider_key = ensure_signed(origin)?;
-			let bounded_name: BoundedVec<u8, T::MaxProviderNameSize> =
-				provider_name.try_into().map_err(|_| Error::<T>::ExceedsMaxProviderNameSize)?;
-
-			let provider_msa_id = Self::ensure_valid_msa_key(&provider_key)?;
-			ProviderToRegistryEntry::<T>::try_mutate(
-				ProviderId(provider_msa_id),
-				|maybe_metadata| -> DispatchResult {
-					ensure!(
-						maybe_metadata.take().is_none(),
-						Error::<T>::DuplicateProviderRegistryEntry
-					);
-					*maybe_metadata = Some(ProviderRegistryEntry { provider_name: bounded_name });
-					Ok(())
-				},
-			)?;
-			Self::deposit_event(Event::ProviderCreated {
-				provider_id: ProviderId(provider_msa_id),
-			});
-			Ok(())
+			let provider_key = T::CreateProviderOrigin::ensure_origin(origin)?;
```

Comment:
if we can only call this function in rococo and local only why use ensure_signed here?

Reply:
I think we can have:

```
pallet_something::Config for Runtime {
....
blah 
blahe
#[feature....]
type CreateProviderOrigin = MsaCreateProviderOrigin;
// The origin that is allowed to create providers via governance
#[feature....]
type CreateProviderViaGovernanceOrigin = MsaCreateProviderViaGovernanceOrigin;
````

Reply:
Fixed